pyof/foundation/network_types.py

```diff
@@ -9,7 +9,102 @@
+    #: ptype (:class:`UBInt16`): The L3 protocol type.
+    ptype = UBInt16()
+    #: hlen (:class:`UBInt8`): Length of hardware address.
+    hlen = UBInt8()
+    #: plen (:class:`UBInt8`): Length of network protocol address.
+    plen = UBInt8()
+    #: operation (:class:`UBInt16`): Determines if it is an ARP request
+    #: or reply.
+    oper = UBInt16()
+    #: sha (:class:`HWAddress`): Sender Hardware Address
+    sha = HWAddress()
+    #: spa (:class:`IPAddress`): Sender Protocol Address
+    spa = IPAddress()
+    #: tha (:class:`HWAddress`): Target Hardware Address
+    tha = HWAddress()
+    #: tpa (:class:`IPAddress`): Target Protocol Address
+    tpa = IPAddress()
+
+    def __init__(self, htype=1, ptype=0x800, hlen=6, plen=4, oper=1,
+                 sha='00:00:00:00:00:00', spa='0.0.0.0',
+                 tha="00:00:00:00:00:00", tpa='0.0.0.0'):
+        """The constructor receives the parameters below.
+
+        Args:
+            htype (int): Hardware protocol type. Defaults to 1 for Ethernet.
+            ptype (int): Network protocol type. Defaults to 0x800 for IPv4.
+            hlen (int): Length of the hardware address. Defaults to 6 for MAC
+                        addresses.
+            plen (int): Length of the networking protocol address. Defaults to
+                        4 for IPv4 addresses.
+            oper (int): Determines the operation for this ARP packet. Must be 1
+                        for ARP request or 2 for ARP reply. Defaults to 1.
+            sha (str): Sender hardware address. Defaults to
+                       '00:00:00:00:00:00'.
+            spa (str): Sender protocol address. Defaults to '0.0.0.0'.
+            tha (str): Target hardware address. Defaults to
+                       '00:00:00:00:00:00'.
+            tpa (str): Target protocol address. Defaults to '0.0.0.0'.
+        """
+        super().__init__()
+        self.htype = htype
+        self.ptype = ptype
+        self.hlen = hlen
+        self.plen = plen
+        self.oper = oper
+        self.sha = sha
+        self.spa = spa
+        self.tha = tha
+        self.tpa = tpa
+
+    def _validate(self):
+        """Assure the ARP contains Ethernet and IPv4 information."""
+        if self.htype.value != 1 or self.ptype.value != 0x800:
+            raise UnpackException("Unsupported protocols in ARP packet")
+        return
+
+    def unpack(self, buff, offset=0):
+        """Unpack a binary struct into this object's attributes.
+
+        Return the values instead of the lib's basic types.
+        Check if the protocols involved are Ethernet and IPv4. Other protocols
+        are currently not supported.
+
+        Args:
+            buff (bytes): Binary buffer.
+            offset (int): Where to begin unpacking.
+
+        Raises:
+            :exc:`~.exceptions.UnpackException`: If unpack fails.
+        """
+        super().unpack(buff, offset)
+        self._validate()
+        self.htype = self.htype.value
```

Comment:
I did it this way to ease the new tutorial's development and following. It is not necessary, as in IPv4, beacuse there's no adjustment to be done for each field.
I can remove it now so we can work in this broader solution, for sure.